packages/node_modules/@webex/internal-plugin-devices/src/devices.js

```diff
@@ -0,0 +1,795 @@
+
+        // Append a ttl value if the device is marked as ephemeral.
+        if (this.config.ephemeral) {
+          body.ttl = this.config.ephemeralDeviceTTL;
+        }
+
+        // Merge header configurations, overriding defaults.
+        const headers = {
+          ...(this.config.defaults.headers ? this.config.defaults.headers : {}),
+          ...(this.config.headers ? this.config.headers : {})
+        };
+
+        return this.request({
+          method: 'PUT',
+          uri: this.url,
+          body,
+          headers
+        })
+          .then((response) => this.processRegistrationSuccess(response))
+          .catch((reason) => {
+            // Handle a 404 error, which indicates that the device is no longer
+            // valid and needs to be registered as a new device.
+            if (reason.statusCode === 404) {
+              this.logger.info('devices: refresh failed, device is not valid');
+              this.logger.info('devices: attempting to register a new device');
+
+              this.clear();
+
+              return this.register();
+            }
+
+            return Promise.reject(reason);
+          });
+      })
+      .catch((error) => {
+        this.logger.warn(
+          'devices: refresh failed, `wdm` service url is not available'
+        );
+
+        return Promise.reject(error);
+      });
+  },
+
+  /**
+   * Register or refresh a device depending on the current device state. Device
+   * registration utilizes the services plugin to send the request to the
+   * **WDM** service.
+   *
+   * @todo
+   * This method contains some logic that could be moved to a `create()` method
+   * at a later time.
+   *
+   * @returns {Promise<void, Error>}
+   */
+  register() {
+    this.logger.info('devices: registering');
+
+    // Validate that the device can be registered.
+    return this.canRegister(true)
+      .then(() => {
+        // Validate if the device is already registered and refresh instead.
+        if (this.registered) {
+          this.logger.info('devices: device already registered, refreshing');
+
+          return this.refresh();
+        }
+
+        // Merge body configurations, overriding defaults.
+        const body = {
+          ...(this.config.defaults.body ? this.config.defaults.body : {}),
+          ...(this.config.body ? this.config.body : {})
+        };
+
+        // Merge header configurations, overriding defaults.
+        const headers = {
+          ...(this.config.defaults.headers ? this.config.defaults.headers : {}),
+          ...(this.config.headers ? this.config.headers : {})
+        };
+
+        // This will be replaced by a `create()` method.
+        return this.request({
+          method: 'POST',
+          service: 'wdm',
+          resource: 'devices',
+          body,
+          headers
+        })
+          .then((response) => this.processRegistrationSuccess(response));
+      })
+      .catch((error) => {
+        this.logger.warn(
```

Comment:
Since we're rejecting, might be good to make this an error.

Reply:
Updated to `error`.